samples/java_layout/src/main/java/io/customer/android/sample/java_layout/ui/core/BaseActivity.java

```diff
@@ -0,0 +1,28 @@
+package io.customer.android.sample.java_layout.ui.core;
+
+import android.os.Bundle;
+
+import androidx.annotation.EmptySuper;
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+import androidx.viewbinding.ViewBinding;
+
+public abstract class BaseActivity<VB extends ViewBinding> extends AppCompatActivity {
```

Comment:
```suggestion
public abstract class BaseActivity<VB extends ViewBinding & Serializable> extends AppCompatActivity {
```

This ensures that the view binding classes used with this activity can be serialized, which is important for Android's state saving mechanism.